src/Credentials/AssumeRoleWithWebIdentityCredentialProvider.php

```diff
@@ -75,32 +83,49 @@ public function __construct(array $config = [])
      */
     public function __invoke()
     {
-        $client = $this->client;
-        try {
-            $token = file_get_contents($this->tokenFile);
-        } catch (\Exception $exception) {
-            throw new CredentialsException(
-                "Error reading WebIdentityTokenFile from " . $this->tokenFile,
-                0,
-                $exception
-            );
-        }
+        return Promise\coroutine(function () {
+            $client = $this->client;
+            $result = null;
+            while ($result == null) {
+                try {
+                    $token = file_get_contents($this->tokenFile);
+                } catch (\Exception $exception) {
+                    throw new CredentialsException(
+                        "Error reading WebIdentityTokenFile from " . $this->tokenFile,
+                        0,
+                        $exception
+                    );
+                }
+
+                $assumeParams = [
+                    'RoleArn' => $this->arn,
+                    'RoleSessionName' => $this->session,
+                    'WebIdentityToken' => $token
+                ];
+
+                try {
+                    $result = $client->assumeRoleWithWebIdentity($assumeParams);
+                } catch (\Exception $e) {
+                    if ($e->getAwsErrorCode() == 'InvalidIdentityToken') {
+                        if ($this->attempts < $this->retries) {
+                            sleep(pow(1.2, $this->attempts));
```

Comment:
Is this retry strategy enough for the cold start scenario we're covering for?

Reply:
A single retry may not succeed when encountering cold starts.  Bumping the default retries to 3 should be sufficient for that scenario.